pkg/kubelet/pleg/evented.go

```diff
@@ -82,11 +83,30 @@ func (e *EventedPLEG) Start() {
 // Healthy check if PLEG work properly.
 // relistThreshold is the maximum interval between two relist.
 func (e *EventedPLEG) Healthy() (bool, error) {
-	return true, nil
-}
+	// TODO(swghosh): relistTime is currently unused
+	// we need getter/setter for relistTime in order to check with
+	// elapsedTime > relistThreshold condition.
+	var relistTime time.Time
+	if relistTimeVal := e.relistTime.Load(); relistTimeVal != nil {
+		relistTime = relistTimeVal.(time.Time)
+	} else {
+		relistTime = time.Time{}
+	}
 
-func (e *EventedPLEG) Relist() {
-	e.genericPLEG.Relist()
+	if relistTime.IsZero() {
+		return false, fmt.Errorf("pleg has yet to be successful")
+	}
+
+	elapsed := e.clock.Since(relistTime)
+	if elapsed > e.relistThreshold {
+		return false, fmt.Errorf("pleg was last seen active %v ago; threshold is %v", elapsed, e.relistThreshold)
+	}
+
+	// PLEG is unhealthy if eventChannel is out of capacity
+	if len(e.eventChannel) >= cap(e.eventChannel) {
+		return false, fmt.Errorf("pleg event channel capacity is full with %v events", len(e.eventChannel))
+	}
+	return true, nil
 }
```

Comment:
Currently, we're missing out on updating `e.relistTime` from the `EventedPLEG` struct, for the first condition to satisfy i.e. `if elapsed > e.relistThreshold` we should do that somewhere.

Reply:
As discussed, I removed `relistPeriod`, `relistTime`, `relistThreshold` from the struct as GenericPLEG will anyway take care of it.

```go
         // GenericPLEG is declared unhealthy when relisting time is more than
	// the relistThreshold. In case, EventedPLEG is turned on the value of
	// relistingPeriod, relistingThreshold is adjusted to higher values and
	// so health of Generic PLEG should take care of the condition checking
	// higher relistThreshold between last two relist.
```